src/decorators.py

```diff
@@ -0,0 +1,46 @@
+from functools import wraps
+
+
+def log(filename=None):
+    """декоратор проверяющий работу функции и записывающий результат в указаный файл или консоль"""
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                result = func(*args, **kwargs)
+                d = f"{func.__name__}: ок"
+                return result, d
+            except Exception as e:
+                d = f"{func.__name__} error: {e}. Inputs: {args}, {kwargs}"
+                return d
```

Comment:
Тут нужно рейзить исключение
```suggestion
                raise e
```